lib/src/webhook.rs

```diff
@@ -0,0 +1,210 @@
+use std::net::SocketAddr;
+
+use futures::{Future, Poll, Stream};
+use futures::future::ok;
+use futures::sync::mpsc::{channel, Receiver, Sender};
+use tokio_core::reactor::Handle;
+
+use telegram_bot_raw::Update;
+use telegram_bot_raw::{DeleteWebhook, SetWebhook};
+
+use serde_json;
+
+use api::Api;
+
+use hyper::error::Error;
+use hyper::{Method, StatusCode};
+use hyper::server::{Http, Request, Response, Service};
+
+const TELEGRAM_WEBHOOK_DEFAULT_PATH: &'static str = "/";
+
+/// This type represents stream of Telegram API updates and uses
+/// webhook under the hood.
+#[must_use = "streams do nothing unless polled"]
+pub struct Webhook {
```

Comment:
Maybe we should split this type into `WebhookStream` and `WebhookConfig`? I want to have type system that doesn't allow streaming without proper registering.